### PR TITLE
doc: Create new chapter about the Dashboard's landing page

### PR DESCRIPTION
**Pull request by ishanrai05**
Fixes: https://tracker.ceph.com/issues/46415

Signed-off-by: Ishan Rai <ishanrai05@gmail.com>

## Checklist
- [x] References tracker ticket
- [x] Updates documentation if necessary
- [x] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by ishanrai05** [comment]
jenkins test docs

File: doc/mgr/dashboard.rst [hunk 1]
```diff
@@ -113,17 +90,67 @@ aspects of your Ceph cluster:
...
+  images and their performance characteristics (read/write ops, traffic).
+  Create, modify and delete iSCSI targets (via ``ceph-iscsi``). Display the
+  iSCSI gateway status on the landing page and info about active initiators.
+  See :ref:`dashboard-iscsi-management` for instructions on how to configure
+  this feature.
+
+Performance
+"""""""""""
+
+* **Client iOPS**:  Display an overview of Input/Output Operations Per Second on
```
Comment (Devp00l) [inline doc/mgr/dashboard.rst:131]:
```suggestion
* **Client IOPS**:  Display an overview of input and output operations per second on
```

File: doc/mgr/dashboard.rst [hunk 2]
```diff
@@ -113,17 +90,67 @@ aspects of your Ceph cluster:
...
+  storage devices the cluster is doing on the client side.
+* **Client Throoughput**: Displays the data transfer rate to and from
+  the storage media in megabytes per second.
+* **CLient READ/Write**: Displays an overview of Read/Write operations performed
```
Comment (Devp00l) [inline doc/mgr/dashboard.rst:135]:
```suggestion
* **CLient Read/Write**: Displays an overview of read and write operations performed
```

Comment (LenzGr) [reply doc/mgr/dashboard.rst:135]: Also note the capital "L" in "CLient"

**Comment by ishanrai05** [comment]
jenkins test dashboard backend

**Comment by ishanrai05** [comment]
jenkins test dashboard backend

**Comment by ishanrai05** [comment]
jenkins test dashboard backend

**Comment by ishanrai05** [comment]
jenkins test dashboard backend

**Comment by ishanrai05** [comment]
jenkins test dashboard

**Comment by ishanrai05** [comment]
jenkins test dashboard backend

**Comment by ishanrai05** [comment]
jenkins test dashboard

**Comment by epuertat** [comment]
jenkins render docs

**Comment by LenzGr** [comment]
retest this please

**Comment by LenzGr** [comment]
Also, since this is a documentation change, please prefix the PR subject and git commit message with `doc:` instead of `mgr/dashboard:`. Thank you!

**Comment by ishanrai05** [comment]
Updated according to #36476

**Comment by ishanrai05** [comment]
jenkins render docs

**Comment by LenzGr** [comment]
Hi Ishan, my previous comment still applies - we're losing some information if you simply move the content of the Dashboard's overall feature overview into a description of the landing page. I think these should be two separate chapters. The first one (the current one) should provide a detailed overview of all of the dashboard's management and monitoring capabilities. Then we should create a chapter that just explains the landing page's widgets and their meanings (which may be somewhat of a repeat of what's listed in the overall feature list).

In a nutshell, there should be two parts: "This is what Ceph Dashboard is capable of" and "This is what the Dashboard landing page displays"

**Comment by ishanrai05** [comment]
jenkins render docs

File: doc/mgr/dashboard.rst [hunk 3]
```diff
@@ -124,6 +124,67 @@ aspects of your Ceph cluster:
...
+Dashboard Landing Page
+^^^^^^^^^^^^^^^^^^^^^^
+Displays overall cluster status, performance and capacity metrics. Gives instant
+feedback to changes in cluster and provides easy access to sub pages of dashboard.
+
+Status
+""""""
+
+* **Cluster Status**: Displays overall cluster health. 
+* **Hosts**: Display a list of all hosts associated to the cluster, which
```
Comment (LenzGr) [inline doc/mgr/dashboard.rst:136]: I'm sorry to intervene again, but I think these descriptions do not accurately describe what's visible on the landing page. Please limit these descriptions to explain the various widgets visible on the landing page only. It's fine to then add a comment that further information is available when clicking on the link that is active on several of the widgets.

Comment (ishanrai05) [reply doc/mgr/dashboard.rst]: I tried following your comment. Please take a look and let me know what you think.

File: doc/mgr/dashboard.rst [hunk 4]
```diff
@@ -124,6 +124,64 @@ aspects of your Ceph cluster:
...
+* **OSDs**: Displays total number of object storage daemons for Ceph (ceph-osds)
+  and the total number of OSDs running (up), total number of OSDs in the cluster
+  (in) and total number of OSDs out of the cluster (out). Provides link to the
+  subpage having list of all the OSDs.
+* **Managers**: Displays the total number of active and standby Ceph Manager
+  daemon (ceph-mgr) running alongside monitor daemons,
+* **Object Gateway**: Displays total number of active object gateways and 
+  provides a link to the subpage displaying list of all the daemons.
+* **Metadata Servers**: Displays total number of active and standby metadata
+  servers daemons (ceph-mds).
```
Comment (LenzGr) [inline doc/mgr/dashboard.rst:151]: I'd suggest to add a reference to "CephFS" here:
```suggestion
  servers daemons for CephFS (ceph-mds).
```

File: doc/mgr/dashboard.rst [hunk 5]
```diff
@@ -124,6 +124,64 @@ aspects of your Ceph cluster:
...
+* **iSCSI Gateways**: Display the total number of iSCSI gateway available, total
```
Comment (LenzGr) [inline doc/mgr/dashboard.rst:152]:
```suggestion
* **iSCSI Gateways**: Display the total number of iSCSI gateways available, total
```

File: doc/mgr/dashboard.rst [hunk 6]
```diff
@@ -124,6 +124,64 @@ aspects of your Ceph cluster:
 * **Ceph Manager Modules**: Enable and disable all Ceph Manager modules, change
   the module-specific configuration settings.
 
+Dashboard Landing Page
```
Comment (LenzGr) [inline doc/mgr/dashboard.rst:127]: How about:
```suggestion
Overview of the Dashboard Landing Page
```

File: doc/mgr/dashboard.rst [hunk 7]
```diff
@@ -124,6 +124,64 @@ aspects of your Ceph cluster:
...
+^^^^^^^^^^^^^^^^^^^^^^
+Displays overall cluster status, performance and capacity metrics. Gives instant
+feedback to changes in cluster and provides easy access to subpages of dashboard.
```
Comment (LenzGr) [inline doc/mgr/dashboard.rst:130]:
```suggestion
feedback to changes in the cluster and provides easy access to subpages of the dashboard.
```

File: doc/mgr/dashboard.rst [hunk 8]
```diff
@@ -124,6 +124,64 @@ aspects of your Ceph cluster:
...
+
+Status
+""""""
+
+* **Cluster Status**: Displays overall cluster health. In case of any error it
+  displays a short description of error and provides a link to the logs.
```
Comment (LenzGr) [inline doc/mgr/dashboard.rst:136]:
```suggestion
  displays a short description of the error and provides a link to the logs.
```

File: doc/mgr/dashboard.rst [hunk 9]
```diff
@@ -124,6 +124,64 @@ aspects of your Ceph cluster:
...
+* **Hosts**: Display the total number of hosts associated to the cluster and
+  link to subpage displaying the list and description of all the hosts.
```
Comment (LenzGr) [inline doc/mgr/dashboard.rst:138]:
```suggestion
  link to the subpage displaying the list and descriptions of all hosts.
```

File: doc/mgr/dashboard.rst [hunk 10]
```diff
@@ -124,6 +124,64 @@ aspects of your Ceph cluster:
...
+* **Monitors**: Displys total number of all MONs and their quorum status, open
+  sessions and link to the subpage having the lists and description of all the
```
Comment (LenzGr) [inline doc/mgr/dashboard.rst:140]:
```suggestion
* **Monitors**: Displays the total number of all MONs and their quorum status, open
  sessions and links to the subpage providing a list and descriptions of all
```

File: doc/mgr/dashboard.rst [hunk 11]
```diff
@@ -124,6 +124,64 @@ aspects of your Ceph cluster:
...
+  MONs.
+* **OSDs**: Displays total number of object storage daemons for Ceph (ceph-osds)
+  and the total number of OSDs running (up), total number of OSDs in the cluster
+  (in) and total number of OSDs out of the cluster (out). Provides link to the
+  subpage having list of all the OSDs.
```
Comment (LenzGr) [inline doc/mgr/dashboard.rst:145]:
```suggestion
  subpage containing a list of all OSDs and related management actions.
```

File: doc/mgr/dashboard.rst [hunk 12]
```diff
@@ -124,6 +124,64 @@ aspects of your Ceph cluster:
...
+* **Managers**: Displays the total number of active and standby Ceph Manager
+  daemon (ceph-mgr) running alongside monitor daemons,
```
Comment (LenzGr) [inline doc/mgr/dashboard.rst:147]:
```suggestion
  daemons (ceph-mgr) running alongside monitor daemons.
```

File: doc/mgr/dashboard.rst [hunk 13]
```diff
@@ -124,6 +124,64 @@ aspects of your Ceph cluster:
...
+* **Object Gateway**: Displays total number of active object gateways and 
+  provides a link to the subpage displaying list of all the daemons.
```
Comment (LenzGr) [inline doc/mgr/dashboard.rst:149]:
```suggestion
* **Object Gateway**: Displays the total number of active object gateways and 
  provides a link to the subpage displaying a list of all object gateway daemons.
```

File: doc/mgr/dashboard.rst [hunk 14]
```diff
@@ -124,6 +124,64 @@ aspects of your Ceph cluster:
...
+* **Metadata Servers**: Displays total number of active and standby metadata
+  servers daemons (ceph-mds).
+* **iSCSI Gateways**: Display the total number of iSCSI gateway available, total
+  number of active iSCSI gateways (up) and total number of inactive iSCSI
+  Gateways (down). Provides link to the subpage having list of all the iSCSI
```
Comment (LenzGr) [inline doc/mgr/dashboard.rst:154]:
```suggestion
  Gateways (down). Provides link to the subpage providing a list of all iSCSI
```

File: doc/mgr/dashboard.rst [hunk 15]
```diff
@@ -124,6 +124,64 @@ aspects of your Ceph cluster:
...
+  Gateways.
+
+Capacity
+""""""""
+
+* **Raw Capacity**: Displays the total physical capacity used out of the total
+  physical capacity provided to the ceph storage nodes (OSDs).
```
Comment (LenzGr) [inline doc/mgr/dashboard.rst:161]:
```suggestion
  physical capacity provided by the ceph storage nodes (OSDs).
```

File: doc/mgr/dashboard.rst [hunk 16]
```diff
@@ -124,6 +124,64 @@ aspects of your Ceph cluster:
...
+* **Objects**: Displays the number of objects in use and the status of objects
+  including the percentage of healthy, misplaced, degraded and unfound status of
+  objects. Object is the smallest unit of data storage in ceph cluster.
```
Comment (LenzGr) [inline doc/mgr/dashboard.rst:164]:
```suggestion
  objects. An object is the smallest unit of data storage in a Ceph cluster.
```

File: doc/mgr/dashboard.rst [hunk 17]
```diff
@@ -124,6 +124,64 @@ aspects of your Ceph cluster:
...
+* **PG Status**: Displays the total number of placement groups in use and the
+  status of the placement group, including the percentage of clean, working,
```
Comment (LenzGr) [inline doc/mgr/dashboard.rst:166]:
```suggestion
  status of placement groups, including the percentage of clean, working,
```

File: doc/mgr/dashboard.rst [hunk 18]
```diff
@@ -124,6 +124,64 @@ aspects of your Ceph cluster:
...
+  warning and unknown status of placement groups.
+* **Pools**: Displays the total number of pools and links to the subpage that
+  lists all Ceph pools and their details.
+* **PGs per OSD**: Displays the number of placement groups per object stotage
```
Comment (LenzGr) [inline doc/mgr/dashboard.rst:170]:
```suggestion
* **PGs per OSD**: Displays the number of placement groups per object storage
```

File: doc/mgr/dashboard.rst [hunk 19]
```diff
@@ -124,6 +124,64 @@ aspects of your Ceph cluster:
...
+  daemons.
+
+Performance
+"""""""""""
+
+* **Client READ/Write**: Displays an overview of rate of read and write operations,
```
Comment (LenzGr) [inline doc/mgr/dashboard.rst:176]:
```suggestion
* **Client READ/Write**: Displays an overview of the rate of read/write operations,
```

File: doc/mgr/dashboard.rst [hunk 20]
```diff
@@ -124,6 +124,64 @@ aspects of your Ceph cluster:
...
+  the total number of input and output operations performed by the cluster on
+  storage devices on the client side.
+* **Client Throughput**: Displays the data transfer rate to and from the storage
+  media.
```
Comment (LenzGr) [inline doc/mgr/dashboard.rst:180]:
```suggestion
* **Client Throughput**: Displays the data transfer rate to and from Ceph clients.
```

File: doc/mgr/dashboard.rst [hunk 21]
```diff
@@ -124,6 +124,64 @@ aspects of your Ceph cluster:
...
+* **Recovery throughput**: Displays rate of moving the data back to the cluster
+  if the cluster is recovering to parity when the hard drive is lost.
+* **Scrubbing**: Displays whether Ceph is comparing the data to other pieces of
+  data to ensure there is no corruption on cluster
```
Comment (LenzGr) [inline doc/mgr/dashboard.rst:184]:
```suggestion
  data to ensure there is no data corruption on the cluster's storage devices.
```

File: doc/mgr/dashboard.rst (hunk 3 above)
Comment (LenzGr) [reply doc/mgr/dashboard.rst]: Looking better, thanks! I left a number of textual improvement suggestions, please take them into consideration.

File: doc/mgr/dashboard.rst [hunk 22]
```diff
@@ -124,6 +124,65 @@ aspects of your Ceph cluster:
...
+feedback to changes in the cluster and provides easy access to subpages of the
+dashboard.
+
+Status
+""""""
+
+* **Cluster Status**: Displays overall cluster health. In case of any error it
+  displays a short description of the error and provides a link to the logs.
+* **Hosts**: Displays the total number of hosts associated to the cluster and
+  link to subpage displaying the list and descriptions of all hosts.
```
Comment (callithea) [inline doc/mgr/dashboard.rst:140]:
```suggestion
  links to the subpage displaying the list and descriptions of all hosts.
```

**Comment by LenzGr** [comment]
jenkins render docs

**Review by LenzGr — APPROVED** [review]
Looks good! Thanks a lot.
